src/gnosisSafe/getWeb3.js

```diff
@@ -0,0 +1,109 @@
+import Web3 from "web3";
+
+export const sameAddress = (firstAddress, secondAddress) => {
+  if (!firstAddress) {
+    return false;
+  }
+
+  if (!secondAddress) {
+    return false;
+  }
+
+  return firstAddress.toLowerCase() === secondAddress.toLowerCase();
+};
+
+export const ETHEREUM_NETWORK = {
+  MAINNET: "MAINNET",
+  MORDEN: "MORDEN",
+  ROPSTEN: "ROPSTEN",
+  RINKEBY: "RINKEBY",
+  GOERLI: "GOERLI",
+  KOVAN: "KOVAN",
+  UNKNOWN: "UNKNOWN",
+};
+
+export const WALLET_PROVIDER = {
+  SAFE: "SAFE",
+  METAMASK: "METAMASK",
+  REMOTE: "REMOTE",
+  TORUS: "TORUS",
+  PORTIS: "PORTIS",
+  FORTMATIC: "FORTMATIC",
+  SQUARELINK: "SQUARELINK",
+  UNILOGIN: "UNILOGIN",
+  WALLETCONNECT: "WALLETCONNECT",
+  OPERA: "OPERA",
+  DAPPER: "DAPPER",
+  WALLETLINK: "WALLETLINK",
+  AUTHEREUM: "AUTHEREUM",
+  LEDGER: "LEDGER",
+  TREZOR: "TREZOR",
+};
+
+export const ETHEREUM_NETWORK_IDS = {
+  1: ETHEREUM_NETWORK.MAINNET,
+  2: ETHEREUM_NETWORK.MORDEN,
+  3: ETHEREUM_NETWORK.ROPSTEN,
+  4: ETHEREUM_NETWORK.RINKEBY,
+  5: ETHEREUM_NETWORK.GOERLI,
+  42: ETHEREUM_NETWORK.KOVAN,
+};
+
+// With some wallets from web3connect you have to use their provider instance only for signing
+// And our own one to fetch data
+export const web3ReadOnly =
+  process.env.NODE_ENV !== "test"
+    ? new Web3(new Web3.providers.HttpProvider(process.env.ETH_NODE_URL))
+    : new Web3(window.web3.currentProvider || "ws://localhost:8545");
+
+let web3 = web3ReadOnly;
+export const getWeb3 = () => web3;
```

Comment:
Can we get this web3 instance from [here](https://github.com/InstaDApp/dsa-sdk/blob/f3d2245489c21deba0afe6e0f562d83ba74960af/src/index.js#L42)?

Reply:
Originally I thought the same. But the problem is many files need to get Web3. And it's hard to pass it the this.web3 object down as other one is just export const function. So this was what Gnosis safe react is using. Or maybe we can change DSA class constructor to use this one as well to make it consistent?